zingolib/src/lightclient.rs

```diff
@@ -946,16 +947,58 @@ impl LightClient {
             "outgoing_metadata" => outgoing_json,
         }
     }
-    /*
-        pub async fn do_list_txsummaries(&self) -> JsonValue {
-            for (txid, transaction_md) in self.wallet.transaction_context.transaction_metadata_set.read().await.current.iter()
-            {
-                let tx_value = transaction_md.total_value_spent();
-                let to_addresses =
-    }
-            Vec<TxSummary>
+    pub async fn do_list_txsummaries(&self) -> Vec<ValueTransferSummary> {
+        let mut sends = Vec::new();
+        let mut receipts = Vec::new();
+
+        for (txid, transaction_md) in self
+            .wallet
+            .transaction_context
+            .transaction_metadata_set
+            .read()
+            .await
+            .current
+            .iter()
+        {
+            let tx_value_spent = transaction_md.total_value_spent();
+            let tx_value_received = transaction_md.total_value_received();
+            let tx_change_received = transaction_md.total_change_returned();
+            match (tx_value_spent, (tx_value_received - tx_change_received)) {
+                //TODO: This is probably an error, if we sent no value and also received no value why
+                //do we have this transaction stored?
+                (0, 0) => unreachable!(),
```

Comment:
Let's start by making this the last statement in the match?

Reply:
Arms are matched top-to-bottom, and it's strictly more-specific than all the other arms, so that would be equivalent to removing it. Which is probably fine, as a) we expect it to be unreachable, and b) the other arms should probably be no-ops if they get a transaction with no value transference in it anyway.